framework/src/modules/interoperability/base_interoperability_module.ts

```diff
@@ -48,12 +57,103 @@ export abstract class BaseInteroperabilityModule extends BaseInteroperableModule
+		// channelData.minReturnFeePerByte == MIN_RETURN_FEE_PER_BYTE_LSK.
+		if (channelData.minReturnFeePerByte !== MIN_RETURN_FEE_PER_BYTE_BEDDOWS) {
+			throw new Error(
+				`channelData.minReturnFeePerByte is not equal to ${MIN_RETURN_FEE_PER_BYTE_BEDDOWS}.`,
+			);
+		}
+	}
+
+	// Commented checks the ones which needs to be checked/applied
+	protected _verifyChainValidators(chainInfo: ChainInfo) {
+		const { chainValidators, chainData } = chainInfo;
+		const { activeValidators, certificateThreshold } = chainValidators;
+
+		// activeValidators must have at least 1 element and at most MAX_NUM_VALIDATORS elements
+		if (activeValidators.length === 0 || activeValidators.length > MAX_NUM_VALIDATORS) {
+			throw new Error(
+				`activeValidators must have at least 1 element and at most ${MAX_NUM_VALIDATORS} elements.`,
+			);
+		}
+
+		// activeValidators must be ordered lexicographically by blsKey property
+		const sortedByBlsKeys = [...activeValidators].sort((a, b) => a.blsKey.compare(b.blsKey));
+		for (let i = 0; i < activeValidators.length; i += 1) {
+			if (!activeValidators[i].blsKey.equals(sortedByBlsKeys[i].blsKey)) {
+				throw new Error('activeValidators must be ordered lexicographically by blsKey property.');
+			}
+		}
+
+		// all blsKey properties must be pairwise distinct
+		const blsKeys = activeValidators.map(v => v.blsKey);
+		if (!objectUtils.bufferArrayUniqueItems(blsKeys)) {
+			throw new Error(`All blsKey properties must be pairwise distinct.`);
+		}
+
+		// for each validator in activeValidators, validator.bftWeight > 0 must hold
+		if (activeValidators.filter(v => v.bftWeight <= 0).length > 0) {
+			throw new Error(`validator.bftWeight must be > 0.`);
+		}
+
+		// let totalWeight be the sum of the bftWeight property of every element in activeValidators.
+		// Then totalWeight has to be less than or equal to MAX_UINT64
+		const totalWeight = activeValidators.reduce(
+			(accumulator, v) => accumulator + v.bftWeight,
+			BigInt(0),
+		);
+		if (totalWeight > MAX_UINT64) {
+			throw new Error(`totalWeight has to be less than or equal to ${MAX_UINT64}.`);
+		}
+
+		// check that totalWeight//3 + 1 <= certificateThreshold <= totalWeight, where // indicates integer division
+		if (
+			totalWeight / BigInt(3) + BigInt(1) > certificateThreshold || // e.g. (300/3) + 1 = 101 > 20
+			certificateThreshold > totalWeight
+		) {
+			throw new Error('Invalid certificateThreshold input.');
+		}
+
+		// check that the corresponding validatorsHash stored in chainInfo.chainData.lastCertificate.validatorsHash
+		// matches with the value computed from activeValidators and certificateThreshold
+		const { validatorsHash } = chainData.lastCertificate;
+		if (!validatorsHash.equals(computeValidatorsHash(activeValidators, certificateThreshold))) {
+			throw new Error('Invalid validatorsHash from chainData.lastCertificate.');
+		}
+	}
+
+	protected _verifyTerminatedStateAccountsCommon(
+		terminatedStateAccounts: TerminatedStateAccountWithChainID[],
```

Comment:
With the new [LIP update PR](https://github.com/LiskHQ/lips/pull/321), two more checks can be added here :

 -  `stateAccount.chainID != getMainchainID()`
 -  `stateAccount.chainId[0] == getMainchainID()[0]`

(see also [this comment in the sidechain module specs](https://github.com/LiskHQ/lisk-sdk/pull/8317/files#r1159574512) where I mention that corresponding checks could be removed from there)

Reply:
No. In the one case you make this check for the `chainID`s stored in `chainInfos` and in the other case for chainIDs stored in `terminatedStateAccounts`, so you need to call this function in two different places. This is why, this should not be removed from LIP

Reply:
Now looks ok, can be resolved from my side.